### ISSUE TITLE
Use TabNet with standard PyTorch model API for data too large to fit in memory

### ISSUE THREAD

**Issue by jlehrer1** [dreamquark-ai/tabnet#378]
## Feature request
Although the sklearn API is quite nice for ease-of-use, it would also be great to use the TabNet model with the standard PyTorch API. 

**What is the expected behavior?**
Call call net = TabNet(100, 10), and use net(sample) and optimizer.backward() to train the model via SGD. 

**What is motivation or use case for adding/changing the behavior?**
There are many cases where writing a manual train loop is preferred, especially when I want to hot-swap this model into an already existing pipeline, or the dataset is too large to fit in memory and can only be accessed sample-wise. This is my entire reason for using TabNet over XGBoost, where creating a dataset distributed in memory is not trivial in certain cases. 

**How should this be implemented in your opinion?**
I see that `pytorch_tabnet.tab_network.TabNet` already exists. What I'm unsure about is the output of the forward pass. It seems to contain both the outputs of a forward pass, as well as `M_loss` defined in the encoder. Should I be using this loss, or a standard `CrossEntropy` loss for classification?

**Are you willing to work on this yourself?**
Yes! This should be a simple thing to do, I just need to know if `M_loss` can be ignored in the output of a forward pass of `pytorch_tabnet.tab_network.TabNet`.

**Comment by Optimox** [comment]
Hello @jlehrer1,

The M_loss part is used to add extra sparsity as you can see here : https://github.com/dreamquark-ai/tabnet/blob/40107a80b0be5ae865d945b85a52e5d99fc19a81/pytorch_tabnet/abstract_model.py#L499

So you can either ignore M_loss and this will be the same as using lambda_sparse=0 or you can add this to your loss in your custom pipeline to add sparsity constraint (as in the original paper).

**Comment by Optimox** [comment]
related to #143

**Comment by jlehrer1** [comment]
@Optimox Gotcha, thank you. Is the correct base model `pytorch_tabnet.tab_model.TabNetClassifier` or should I be using `pytorch_tabnet.tab_network.TabNet` as my PyTorch equivalent to the sklearn API wrapper?

**Comment by jlehrer1** [comment]
It would be really nice to use the base model with raw `PyTorch`, as in having the usual `class Model(torch.nn.Module)`.

**Comment by jlehrer1** [comment]
Perfect, thanks. I wasn't sure if there was any extra logic outside of `tab_network.py` besides adding the `M_loss`, but it seems there isn't (besides the explainability).
